compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -256,13 +256,13 @@ static LogicalResult setContractConfig(func::FuncOp entryPoint,
 
         SmallVector<unsigned> partitionedLoops =
             cast<PartitionableLoopsInterface>(op.getOperation())
-                .getPartitionableLoops(kNumMaxParallelDims);
+                .getPartitionableLoops(/*maxNumPartitionedLoops=*/5);
```

Comment:
What would happen if you just drop this.

Reply:
I couldn't find an implementation of it that allows 0 arguments. 
![Screenshot 2023-05-04 at 7 14 43 PM](https://user-images.githubusercontent.com/1041731/236349525-739b6466-5e89-40c1-801b-3756e2cdf2ae.png)

Reply:
Just pass `std::nullopt` . That is like setting it to no limit (should be easy to see from the implementation)

Reply:
This comment isnt addressed yet. If you pass `std::nullopt`, the number of parallel dimensions isnt pruned and all the parallel dimensions are returned. (See https://github.com/openxla/iree/blob/8aa35a44551563f79559724151829050a2e6e3d4/compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp#L44 )

Reply:
Please add a test for this case.